core/src/main/java/org/apache/iceberg/stats/Blob.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+
+public final class Blob {
+  private final String type;
+  private final List<Integer> inputFields;
+  private final ByteBuffer blobData;
+  private final StatsCompressionCodec requestedCompression;
+
+  public Blob(String type, List<Integer> inputFields, ByteBuffer blobData) {
+    this(type, inputFields, blobData, null);
+  }
+
+  public Blob(
+      String type, List<Integer> inputFields, ByteBuffer blobData,
+      @Nullable StatsCompressionCodec requestedCompression) {
```

Comment:
Is this still `@Nullable` now that there is a `NONE` symbol?

Reply:
It is. When not set, the writer's configured default compression is applied, per https://github.com/apache/iceberg/pull/4537#discussion_r871523692